[0.3] SEGV when trying to create a new cluster

/kind bug

**What steps did you take and what happened:**
Create a new cluster using the management cluster workflow, the actuator throws a panic with the following:
```
I0723 23:42:46.184838       1 plugins.go:39] Registered cluster provisioner "aws"
I0723 23:42:46.186520       1 reflector.go:131] Starting reflector *v1alpha1.Cluster (10m0s) from pkg/cache/internal/informers_map.go:126
I0723 23:42:46.186585       1 reflector.go:169] Listing and watching *v1alpha1.Cluster from pkg/cache/internal/informers_map.go:126
I0723 23:42:46.187453       1 reflector.go:131] Starting reflector *v1alpha1.Machine (10m0s) from pkg/cache/internal/informers_map.go:126
I0723 23:42:46.187485       1 reflector.go:169] Listing and watching *v1alpha1.Machine from pkg/cache/internal/informers_map.go:126
I0723 23:42:46.386069       1 cluster_controller.go:92] Running reconcile Cluster for "test1"
I0723 23:42:46.386229       1 cluster_controller.go:160] reconciling cluster object test1 triggers idempotent reconcile.
I0723 23:42:46.386350       1 actuator.go:79] [cluster-actuator] "level"=0 "msg"="Reconciling Cluster" "cluster-name"="test1" "cluster-namespace"="default"
I0723 23:42:46.386351       1 machine_controller.go:115] Reconciling Machine "test1-controlplane-0"
I0723 23:42:46.386465       1 actuator.go:418] [machine-actuator] "level"=0 "msg"="Checking if machine exists in cluster"  "cluster-name"="test1" "machine-name"="test1-controlplane-0" "machine-namespace"="default"
I0723 23:42:46.387084       1 certificates.go:72] [cluster-actuator]/cluster.k8s.io/v1alpha1/default/test1 "level"=2 "msg"="Reconciling certificates"  "cluster-name"="test1" "cluster-namespace"="default"
I0723 23:42:46.387118       1 certificates.go:75] [cluster-actuator]/cluster.k8s.io/v1alpha1/default/test1 "level"=2 "msg"="Generating keypair for"  "user"="cluster-ca"
E0723 23:42:46.387358       1 runtime.go:69] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76
vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
GOROOT/src/runtime/panic.go:522
GOROOT/src/runtime/panic.go:82
GOROOT/src/runtime/signal_unix.go:390
pkg/cloud/aws/actuators/machine_scope.go:131
pkg/cloud/aws/actuators/machine/actuator.go:431
vendor/sigs.k8s.io/cluster-api/pkg/controller/machine/machine_controller.go:197
vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215
vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158
vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
src/runtime/asm_amd64.s:1337
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12de9cf]

goroutine 184 [running]:
sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x105
panic(0x1721180, 0x2cec9e0)
	GOROOT/src/runtime/panic.go:522 +0x1b5
sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/actuators.(*MachineScope).Close(0xc00015f940)
	pkg/cloud/aws/actuators/machine_scope.go:131 +0x8f
sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/actuators/machine.(*Actuator).Exists(0xc0003dabe0, 0x1c47940, 0xc0000b0060, 0xc000451860, 0xc0000f9080, 0x0, 0x0, 0x0)
	pkg/cloud/aws/actuators/machine/actuator.go:431 +0x988
sigs.k8s.io/cluster-api-provider-aws/vendor/sigs.k8s.io/cluster-api/pkg/controller/machine.(*ReconcileMachine).Reconcile(0xc00003af40, 0xc00014ab69, 0x7, 0xc0002b52a0, 0x14, 0x2d04a60, 0xc00029c1b0, 0xc000056e18, 0x144327c)
	vendor/sigs.k8s.io/cluster-api/pkg/controller/machine/machine_controller.go:197 +0x379
sigs.k8s.io/cluster-api-provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00046c3c0, 0x0)
	vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215 +0x1cc
sigs.k8s.io/cluster-api-provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1()
	vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158 +0x36
sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc00029c1e0)
	vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00029c1e0, 0x3b9aca00, 0x0, 0x1, 0xc0004cb140)
	vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xf8
sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc00029c1e0, 0x3b9aca00, 0xc0004cb140)
	vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/cluster-api-provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:157 +0x311
```

Which points to ProviderID in `MachineScope.Close` that could be nil in case the instance hasn't been created yet. This situation can happen for example when the cluster isn't ready yet.


**What did you expect to happen:**
Manager should not panic

--- a/pkg/cloud/aws/actuators/cluster/actuator.go
+++ b/pkg/cloud/aws/actuators/cluster/actuator.go
@@ -123,6 +123,14 @@ func (a *Actuator) Reconcile(cluster *clusterv1.Cluster) error {
 		kubeConfigSecret := &apiv1.Secret{
 			ObjectMeta: metav1.ObjectMeta{
 				Name: kubeConfigSecretName,
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion: cluster.APIVersion,
+						Kind:       cluster.Kind,
+						Name:       cluster.Name,
+						UID:        cluster.UID,
+					},
+				},
 			},
 			StringData: map[string]string{
 				"value": kubeConfig,

--- a/pkg/cloud/aws/actuators/machine_scope.go
+++ b/pkg/cloud/aws/actuators/machine_scope.go
@@ -127,7 +127,7 @@ func (m *MachineScope) Close() {
 		return
 	}
 
-	if m.Machine.Spec.ProviderID == nil || *m.Machine.Spec.ProviderID == "" {
+	if m.MachineStatus.InstanceID != nil && (m.Machine.Spec.ProviderID == nil || *m.Machine.Spec.ProviderID == "") {
 		providerID := fmt.Sprintf("aws:////%s", *m.MachineStatus.InstanceID)
 		m.Machine.Spec.ProviderID = &providerID
 	}

